### PR TITLE
Add Bastion native client support

### PR DESCRIPTION
**Pull request by juang903**
# Overview/Summary

Adds the option to enable Bastion native client support. Defaults to disabled.

### CODE REVIEW

File: infra-as-code/bicep/modules/hubNetworking/hubNetworking.bicep [hunk 1]
```diff
@@ -70,6 +70,9 @@ param parAzBastionName string = '${parCompanyPrefix}-bastion'
 ])
 param parAzBastionSku string = 'Standard'
 
+@sys.description('Switch to enable/disable Bastion native client support.')
+param parBastionTunneling bool = false
```
Comment (jtracey93) [inline infra-as-code/bicep/modules/hubNetworking/hubNetworking.bicep:74]:
```suggestion
param parAzBastionTunneling bool = false
```
making it match the others

File: infra-as-code/bicep/modules/hubNetworking/hubNetworking.bicep [hunk 2]
```diff
@@ -526,6 +529,7 @@ resource resBastion 'Microsoft.Network/bastionHosts@2023-02-01' = if (parAzBasti
   }
   properties: {
     dnsName: uniqueString(resourceGroup().id)
+    enableTunneling: parBastionTunneling
```
Comment (jtracey93) [inline infra-as-code/bicep/modules/hubNetworking/hubNetworking.bicep:532]:
```suggestion
    enableTunneling: (parAzBastionSku == 'Standard' && parAzBastionTunneling) ? parAzBastionTunneling : false
```

adding some logic for ensuring standard has been selected 👍

Can you please test both deploy with standard and basic to ensure all works

File: infra-as-code/bicep/modules/hubNetworking/hubNetworking.bicep [hunk 3]
```diff
@@ -70,6 +70,9 @@ param parAzBastionName string = '${parCompanyPrefix}-bastion'
 ])
 param parAzBastionSku string = 'Standard'
 
+@sys.description('Switch to enable/disable Bastion native client support.')
```
Comment (jtracey93) [inline infra-as-code/bicep/modules/hubNetworking/hubNetworking.bicep:73]:
```suggestion
@sys.description('Switch to enable/disable Bastion native client support. This is only supported when the Standard SKU is used for Bastion as documented here: https://learn.microsoft.com/azure/bastion/native-client')
```

**Review by jtracey93 — CHANGES_REQUESTED** [review]
Thanks @juang903 for the PR, please see the PR comments and also can you manually generate the doc updates by following this please https://github.com/Azure/ALZ-Bicep/wiki/Contributing#manually-generating-the-parameter-markdown-files

Thanks 

Jack

**Comment by juang903** [comment]
@microsoft-github-policy-service agree

**Comment by juang903** [comment]
Hey @jtracey93, updated with changes requested and markdown files.

**Review by jtracey93 — APPROVED** [review]
LGTM

**Pull request merged**